lib/ipam/ipam.go

```diff
@@ -1124,12 +1125,15 @@ func (c ipamClient) ReleaseHostAffinities(ctx context.Context, host string, must
 				if _, ok := err.(errBlockClaimConflict); ok {
 					// Claimed by a different host.
 				} else {
-					return err
+					// Store the error for later so we can return it.
+					// We don't want to return just yet so we can do a best-effort
+					// attempt at releasing the other CIDRs for this host.
+					storedError = err
```

Comment:
It may not be important, but there's a slight change in what error is returned. If multiple blocks would have failed, the previous code would return the error for the first (and only) block that failed. This code returns the error for the last block that failed.

Also, do you think it would be worth it to either aggregate the errors into one error, or return multiple list of errors?

Reply:
Yeah, I considered this. Overall I figured it was unimportant because I don't think any of the calling code cares about _which_ block actually failed, and actually can't reliably rely on that anyway.

IMO logging out the intermediate errors is enough (they'll show up in the log) and we'll just return the final error to indicate the operation failed.